File: 3367-find-the-sum-of-encrypted-integers/find-the-sum-of-encrypted-integers.py
class Solution(object):
    def sumOfEncryptedInt(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        def findMax(n):
            mx=0
            while n != 0:
                rem=n%10
                mx=max(mx,rem)
                n//=10
            return mx
        for i in range(len(nums)):
            mx=findMax(nums[i])
            n=len(str(nums[i]))
            nums[i]=int(str(mx)*n)
        return sum(nums)

File: 3367-find-the-sum-of-encrypted-integers/find-the-sum-of-encrypted-integers.java
class Solution {
    public int sumOfEncryptedInt(int[] nums) {
        for(int i=0;i<nums.length;i++){
            int max = findMax(nums[i]);
            int n=String.valueOf(nums[i]).length();
            
            nums[i]=Integer.valueOf(String.valueOf(max).repeat(n));
        }
        return Arrays.stream(nums).sum();
    }
    public int findMax(int n){
        int max=0;
        while(n!=0){
            int rem=n%10;
            max=Math.max(rem,max);
            n=n/10;
        }
        return max;
    }
}